src/main/java/io/debezium/connector/spanner/db/model/schema/SpannerSchema.java

```diff
@@ -51,9 +51,19 @@ public void addColumn(String tableName, String columnName, String type, long ord
                 columns = new ArrayList<>();
                 tableMap.put(tableName, columns);
             }
-            if (!columns.stream().anyMatch(column -> column.getName().equals(columnName))) {
-                columns.add(Column.create(columnName, type, primaryKey, ordinalPosition, nullable));
+            columns.add(Column.create(columnName, type, tableToPrimaryColumnMap.get(tableName).contains(columnName), ordinalPosition, nullable));
+        }
+
+        public void addPrimaryColumn(String tableName, String columnName) {
+            List<String> primaryColumns;
+            if (tableToPrimaryColumnMap.containsKey(tableName)) {
+                primaryColumns = tableToPrimaryColumnMap.get(tableName);
+            }
+            else {
+                primaryColumns = new ArrayList<>();
+                tableToPrimaryColumnMap.put(tableName, primaryColumns);
             }
+            primaryColumns.add(columnName);
```

Comment:
What's the point of having this list here?

Also why would we benefit from using an application level join rather than DB join?

Reply:
The list here only contains primary columns of a table.

DB level left join return duplicate columns there was a issue related to that which was fixed in this PR https://github.com/debezium/debezium-connector-spanner/pull/18. 
In the PR we anyways had to put application level processing to filter out duplicate columns. 

The issue which current PR solves is that when a secondary index is made on column which is also a primary key, an error is thrown as mentioned in jira ticket. To prevent this at db level, query will become complex. 

What does Application level join mean:-
1. Fetch primary columns of a table separately using **information_schema.index_columns** table of db. 
2. Fetch columns info of the table from **information_schema.columns** table of db. 
3. To find if the column is primary key, use the code logic. The purpose of join was only to find primary column.